src/main/java/org/folio/linked/data/util/BibframeUtils.java

```diff
@@ -103,4 +106,15 @@ public static List<Resource> extractInstancesFromWork(Resource resource) {
       .toList();
   }
 
+  public static Resource ensureActive(Resource resource, ResourceRepository repository) {
+    if (resource.isActive()) {
+      return resource;
+    }
+    return ofNullable(resource.getFolioMetadata())
```

Comment:
I think this logic will not work any more as `resource.getFolioMetadata()` will be null for non active resource.
I think this is what we need to do

Check if there is REDIRECT outgoing edge from current resource.
1. If yes, recursively call `return ensureActive(redirectedResource, repository);`
2. If not, `return resource` (even if the resource is inactive, we do not have a choice. So, return the inactive resource. I do not think this scenario will happen in real-life )

Reply:
oh yeah, you're right. Fixed, thank you!